File: jvm-experiments-timeperiod/src/main/java/kr/kth/timeperiod/ITimePeriodChain.java
package kr.kth.timeperiod;

/**
 * {@link ITimePeriod}의 Linked List 스타일의 컨테이너입니다.
 * User: sunghyouk.bae@gmail.com
 * Date: 12. 9. 18
 */
public interface ITimePeriodChain extends ITimePeriodContainer {

    /**
     * 첫번째 항목
     */
    ITimePeriod getFirst();

    /**
     * 마지막 항목
     */
    ITimePeriod getLast();
}

File: jvm-experiments-timeperiod/src/main/scala/kr/kth/timeperiod/ITimePeriodChain.scala
package kr.kth.timeperiod

/**
 * kr.kth.timeperiod.ITimePeriodChain
 * User: sunghyouk.bae@gmail.com
 * Date: 13. 1. 14
 */
trait ITimePeriodChain extends ITimePeriodContainer {
  /**
   * 첫번째 항목
   */
  def getFirst: ITimePeriod =
    if (size > 0) get(0)
    else null

  /**
   * 마지막 항목
   */
  def getLast: ITimePeriod =
    if (size > 0) get(size - 1)
    else null
}
